lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -1521,4 +1521,37 @@ namespace Js
 
         return NumberUtilities::TryToInt64(length);
     }
+
+    Var JavascriptConversion::TryCanonicalizeAsSimpleVar(Var value)
+    {
+        TypeId typeId = JavascriptOperators::GetTypeId(value);
+        switch (typeId)
+        {
+        case TypeIds_Integer:
+        case TypeIds_Number:
+        {
+            Var taggedInt = TryCanonicalizeAsTaggedInt<true>(value);
+            if (taggedInt)
+            {
+                return taggedInt;
+            }
+
+#if FLOATVAR
+            return value;
+#else
+            return nullptr;
+#endif
+        }
+        case TypeIds_String:
+        case TypeIds_Symbol:
```

Comment:
>TypeIds_Symbol [](start = 13, length = 14)

It looks like there's also an existing problem where SameValueComparer doesn't have any special case for Symbols and compares them by reference. Would you like to fix that while you're working on related things, or should I open a separate bug?

Reply:
sure I can fix it.